tests/cpp/test_host_ir_stream_lowering.cpp

```diff
@@ -0,0 +1,823 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2025-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <gtest/gtest.h>
+
+#include <fusion.h>
+#include <host_ir/container.h>
+#include <host_ir/executor.h>
+#include <host_ir/lower.h>
+#include <ir/all_nodes.h>
+#include <ir/builder.h>
+#include <kernel_ir.h>
+#include <multidevice/executor.h>
+#include <ops/all_ops.h>
+#include <preseg_passes/stream_parallel_type.h>
+#include <tests/cpp/utils.h>
+
+#include <algorithm>
+#include <iostream>
+
+namespace nvfuser {
+
+namespace hir {
+
+using HirLowerStreamTest = NVFuserTest;
+
+TEST_F(HirLowerStreamTest, InputsAreNotStreamParallelized) {
```

Comment:
I suggest to start the review by inspecting those test and by running them with the option `NVFUSER_DUMP=host_ir`.

There are two identical sets of tests: `HirLowerStreamTest` and `MultiDeviceExecutorLowerStreamTest`. Both test semantically identical situation, but the former is "lower level" in the sense that it manually builds a Hir container  and apply the pass ; while the latter goes through the main level API for defining a fusion, which then gets automatically lowered and executed through MultiDeviceExecutor